File: cpp-tiny-virtual-machine/cmd.cpp
#include "iostream"
#include "src/machine.h"
using namespace std;
int main()
{
  Machine *machine = new Machine();
  string inputCodeLine = "";
  cout << ">";
  while (getline(cin, inputCodeLine))
  {
    if (inputCodeLine.find("exit") != string::npos)
      return 0;
    machine->process(inputCodeLine);
    cout << ">";
  }
  return 0;
}

File: proto-virtual-machine/cmd.ts
import { createInterface } from 'readline';
import VitrulMachine from "./src/machine"
const machine = new VitrulMachine();

const readline = createInterface({
  input: process.stdin,
  output: process.stdout,
  prompt: '>'
})
readline.prompt();

readline.on('line', (line) => {
  const lineCode = line.trim();
  if (lineCode == "exit()") {
    console.log('再见!');
    process.exit(0);
    return;
  }
  if (lineCode == "context") {
    console.log(machine.context);
    return;
  }

  machine.process(lineCode);
  prompt();
}).on('close', () => {
  console.log('再见!');
  process.exit(0);
});
